ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/AdminResponseHandlers.java

```diff
@@ -98,6 +101,18 @@ static void handleDescribeSourceResponse(
     }
   }
 
+  static void handleServerInfoResponse(
+      final JsonObject serverInfoEntity,
+      final CompletableFuture<ServerInfo> cf
+  ) {
+    final JsonObject source = serverInfoEntity.getJsonObject("KsqlServerInfo");
+    cf.complete(new ServerInfoImpl(
```

Comment:
Let's wrap the creation of the `ServerInfoImpl` in a try-catch block, so we can fail the future with a more informative error message if something goes wrong (nothing should go wrong today, but it'd be good to future-proof in the event that the response format is inadvertently changed in an incompatible way).